assets/migrations/postgres/005_drop_indices.sql

```diff
@@ -0,0 +1,40 @@
+-- +goose NO TRANSACTION
+-- This migration updates the primary key of the tuple table, drops old indexes
+
+-- +goose Up
+
+ALTER TABLE tuple
```

Comment:
To do this with nearly 0 locking and reduce interruptions to existing queries you'll need to first build a unique index concurrently and then replace the primary key using that index.

For example:

```sql
CREATE UNIQUE INDEX CONCURRENTLY idx_tuple_pkey
   ON tuple (store, object_type, object_id, relation, user_object_type, user_object_id, user_relation);

ALTER TABLE tuple
   ADD CONSTRAINT tuple_pkey PRIMARY KEY USING INDEX idx_tuple_pkey;
```

The `ALTER TABLE .. ADD CONSTRAINT .. PRIMARY KEY` will lock the whole table, but the constraint swaps over to the index immediately and the indexing continues concurrently. So the effect to table/row level locks is much more negligible than the alternative.